src/runtime_intrinsics.c

```diff
@@ -806,7 +806,7 @@ bi_iintrinsic_fast(LLVMXor, xor_op, xor_int, u)
 bi_iintrinsic_cnvtb_fast(LLVMShl, shl_op, shl_int, u, 1)
 #define lshr_op(a,b) (b >= 8 * sizeof(a)) ? 0 : a >> b
 bi_iintrinsic_cnvtb_fast(LLVMLShr, lshr_op, lshr_int, u, 1)
-#define ashr_op(a,b) (b < 0 || b >= 8 * sizeof(a) ? 0 : a >> b)
+#define ashr_op(a,b) (b < 0 ? 0 : b >= 8 * sizeof(a) ? a >> (8*sizeof(a) - 1) : a >> b)
```

Comment:
(-1) << (-1) == -1, not zero

also, you can just use `~0` as a more concise way to write the result